dbms/src/Storages/MergeTree/MergeTreeReadPool.cpp

```diff
@@ -251,6 +240,10 @@ void MergeTreeReadPool::fillPerThreadInfo(
 {
     threads_tasks.resize(threads);
 
+    /// Let the ranges be listed from right to left so that the leftmost range can be dropped using `pop_back()`.
```

Comment:
If the only reason to reverse the array is to have O(1) deletion, we can just use `std::deque`.

Reply:
Yes, we can. Actually I don't why originally std::vector was used.